ui/app/src/state/reducers/pathNavigatorTree.ts

```diff
@@ -0,0 +1,164 @@
+          keywordByPath,
+          totalByPath: {}
+        }
+      };
+    },
+    [pathNavigatorTreeExpandPath.type]: (state, { payload: { id, path } }) => {
+      return {
+        ...state,
+        [id]: {
+          ...state[id],
+          expanded: [...state[id].expanded, path]
+        }
+      };
+    },
+    [pathNavigatorTreeCollapsePath.type]: (state, { payload: { id, path } }) => {
+      return {
+        ...state,
+        [id]: {
+          ...state[id],
+          expanded: state[id].expanded.filter((expanded) => expanded !== path)
+        }
+      };
+    },
+    [pathNavigatorTreeToggleExpanded.type]: (state, { payload: { id, collapsed } }) => {
+      return {
+        ...state,
+        [id]: {
+          ...state[id],
+          collapsed
+        }
+      };
+    },
+    [pathNavigatorTreeSetKeyword.type]: (state, { payload: { id, path, keyword } }) => {
+      return {
+        ...state,
+        [id]: {
+          ...state[id],
+          keywordByPath: {
+            ...state[id].keywordByPath,
+            [path]: keyword
+          }
+        }
+      };
+    },
+    [pathNavigatorTreeFetchPathChildren.type]: (state, { payload: { id, path } }) => {
+      return {
+        ...state,
+        [id]: {
+          ...state[id],
+          ...(!state[id].expanded.includes(path) && { expanded: [...state[id].expanded, path] })
+        }
+      };
+    },
+    [pathNavigatorTreeFetchPathChildrenComplete.type]: (state, { payload: { id, parentPath, children, options } }) => {
+      return {
+        ...state,
+        [id]: {
+          ...state[id],
+          // if the expanded node has not children and is not a result from a filtering it means is a leaf node
+          ...(children.length === 0 &&
+            !options?.keyword && { expanded: state[id].expanded.filter((path) => path !== parentPath) }),
+          childrenByParentPath: {
+            ...state[id].childrenByParentPath,
+            [parentPath]: children.map((item) => item.path)
+          },
+          totalByPath: {
+            ...state[id].totalByPath,
+            [parentPath]: children.total
+          }
+        }
+      };
+    },
+    [pathNavigatorTreeFetchPathPageComplete.type]: (state, { payload: { id, parentPath, children, options } }) => {
+      return {
+        ...state,
+        [id]: {
+          ...state[id],
+          childrenByParentPath: {
+            ...state[id].childrenByParentPath,
+            [parentPath]: [...state[id].childrenByParentPath[parentPath], ...children.map((item) => item.path)]
+          },
+          totalByPath: {
+            ...state[id].totalByPath,
+            [parentPath]: children.total
+          }
+        }
+      };
+    },
+    [pathNavigatorTreeRestoreComplete.type]: (state, { payload: { id, data } }) => {
+      const children = {};
+      const total = {};
+      Object.keys(data).forEach((path) => {
+        children[path] = [...data[path].map((item) => item.path)];
```

Comment:
Map returns a new array.
```suggestion
        children[path] = data[path].map((item) => item.path);
```